pkg/activator/net/revision_backends.go

```diff
@@ -259,30 +259,46 @@ func (rw *revisionWatcher) checkDests(dests []string) {
 	}
 }
 
-func (rw *revisionWatcher) runWithTickCh(tickCh <-chan time.Time) {
+func (rw *revisionWatcher) run(probeFrequency time.Duration) {
 	var dests []string
+
 	for {
-		select {
-		case <-rw.doneCh:
-			return
-		case x, ok := <-rw.destsChan:
-			if !ok {
+		// We dont need to tick for probes if we have no dests or we are in clusterIP mode
+		if dests == nil || rw.clusterIPHealthy {
```

Comment:
Does clusterIPHealthy need to be guarded?

Reply:
Nope - we only access it in checkDests() which is only called from this event loop